File: go-ut-bench/datasets/java/java_code_files_self_contained/interface_mock/interface_mock_001.java
interface Logger {
    void log(String message);
    String[] getLogs();
}

class ConsoleLogger implements Logger {
    private String[] logs;
    private int count;
    
    public ConsoleLogger() {
        this.logs = new String[100];
        this.count = 0;
    }
    
    public void log(String message) {
        if (count < logs.length) {
            logs[count++] = message;
        }
    }
    
    public String[] getLogs() {
        String[] result = new String[count];
        for (int i = 0; i < count; i++) {
            result[i] = logs[i];
        }
        return result;
    }
}

class FileLogger implements Logger {
    private String filename;
    private String[] logs;
    private int count;
    
    public FileLogger(String filename) {
        this.filename = filename;
        this.logs = new String[100];
        this.count = 0;
    }
    
    public void log(String message) {
        if (count < logs.length) {
            logs[count++] = "[" + filename + "] " + message;
        }
    }
    
    public String[] getLogs() {
        String[] result = new String[count];
        for (int i = 0; i < count; i++) {
            result[i] = logs[i];
        }
        return result;
    }
}

class Application {
    private Logger logger;
    
    public Application(Logger logger) {
        this.logger = logger;
    }
    
    public String run(String taskName) {
        logger.log("Starting task: " + taskName);
        String result = "Task " + taskName + " completed";
        logger.log(result);
        return result;
    }
    
    public String[] getHistory() {
        return logger.getLogs();
    }
}

File: go-ut-bench/datasets/cpp/cpp_code_files_self_contained/interface_mock/interface_mock_001.cpp
#include <vector>
#include <string>
#include <stdexcept>

class Logger {
public:
    virtual void log(const std::string& message) = 0;
    virtual std::vector<std::string> getLogs() = 0;
    virtual ~Logger() {}
};

class ConsoleLogger : public Logger {
private:
    std::vector<std::string> logs;
    
public:
    ConsoleLogger() : logs() {}
    
    void log(const std::string& message) override {
        logs.push_back(message);
    }
    
    std::vector<std::string> getLogs() override {
        return logs;
    }
};

class FileLogger : public Logger {
private:
    std::string filename;
    std::vector<std::string> logs;
    
public:
    FileLogger(const std::string& name) : filename(name), logs() {}
    
    void log(const std::string& message) override {
        logs.push_back("[" + filename + "] " + message);
    }
    
    std::vector<std::string> getLogs() override {
        return logs;
    }
};

class Application {
private:
    Logger* logger;
    
public:
    Application(Logger* log) : logger(log) {}
    
    std::string run(const std::string& taskName) {
        logger->log("Starting task: " + taskName);
        std::string result = "Task " + taskName + " completed";
        logger->log(result);
        return result;
    }
    
    std::vector<std::string> getHistory() {
        return logger->getLogs();
    }
};